### PR TITLE
Simpler syntax for overriding modules

### PR DESCRIPTION
**Pull request by paulcsmith**

### CODE REVIEW

File: lib/pact.ex [hunk 1]
```diff
@@ -43,11 +43,18 @@ defmodule Pact do
...
   @doc "Override all calls to `name` in `pid` with `module`"
-  def override(pid, name, module) do
+  def override(pid, name, module) when is_atom(module) do
     name = to_string(name)
     GenServer.cast(__MODULE__, {:override, pid, name, module})
   end
 
+  def override(pid, name, overridden_functions) when is_list(overridden_functions) do
+    name = to_string(name)
+    {:module, fake_module_name, _, _} = create_fake_module(Pact.Fake)
```
Comment (paulcsmith) [inline lib/pact.ex]: I need to figure out a way to dynamically create a module name here so that there are no collisions

File: lib/pact.ex [hunk 2]
```diff
@@ -110,6 +120,21 @@ defmodule Pact do
     {:stop, :normal, :ok, container}
   end
 
+  defp create_fake_module_name(name, pid) do
+    process_id_as_string = IO.iodata_to_binary(:erlang.pid_to_list(pid))
```
Comment (BlakeWilliams) [inline lib/pact.ex]: Whaaaaaaat? What is this doing?

File: lib/pact.ex [hunk 3]
```diff
@@ -110,6 +120,21 @@ defmodule Pact do
...
+    module_name = String.capitalize(name)
+    String.to_atom("PactFakes#{module_name}#{process_id_as_string}")
+  end
+
+  defp stub_module(module, overridden_functions) do
+    :meck.new(module, [:no_link, :non_strict])
```
Comment (BlakeWilliams) [inline lib/pact.ex:142]: I'm curious, could we replace this with `Module.create`?

```
quote do
  module.create(
    unquote(name),
    Enum.each(overridden_functions, fn ({name, function}) ->
      def unquote(name) do: unquote(function)
    end)
end
```

or something similar?

File: lib/pact.ex [hunk 4]
```diff
@@ -110,6 +120,21 @@ defmodule Pact do
...
+    Enum.each(overridden_functions, fn (function_pair) ->
```
Comment (BlakeWilliams) [inline lib/pact.ex]: You can pattern match out `function_name` and `function` here.

File: lib/pact.ex (hunk 2 above)
Comment (paulcsmith) [inline lib/pact.ex]: That's how you get the id of the pid. It's ripped from the `IO.inspect` for `PID`. You can't pass a process to `to_string` either :(

Comment (BlakeWilliams) [inline lib/pact.ex]: That's unfortunate!

Comment (BlakeWilliams) [inline lib/pact.ex]: I dislike comments, but is it worth one here?

File: lib/pact.ex (hunk 4 above)
Comment (paulcsmith) [inline lib/pact.ex]: I'm not sure what you mean. Can you paste a code sample?

File: lib/pact.ex (hunk 3 above)
Comment (paulcsmith) [inline lib/pact.ex:142]: Definitely could be done. Would you prefer that over using meck?

File: lib/pact.ex (hunk 2 above)
Comment (paulcsmith) [inline lib/pact.ex]: I thought about that too, but I'm not sure. Is the variable name not clear enough? Maybe I could extract a function to make it clear?

File: lib/pact.ex (hunk 3 above)
Comment (BlakeWilliams) [inline lib/pact.ex:142]: It's not much more code, and it saves us a dependency so I lean towards yes. Your thoughts?

Comment (paulcsmith) [inline lib/pact.ex:142]: I'll give it a shot :) If it works well, I'm cool with it

Comment (BlakeWilliams) [inline lib/pact.ex:142]: Awesome, if you get stuck ping me on Slack. I think it'd be fun to work on!

File: lib/pact.ex (hunk 4 above)
Comment (BlakeWilliams) [inline lib/pact.ex]: `fn ({function_name, function})`

**Pull request merged**